sycl/source/detail/program_manager/program_manager.hpp

```diff
@@ -134,6 +135,29 @@ class ProgramManager {
                              const device &Device, pi::PiProgram NativePrg,
                              const string_class &KernelName, bool KnownProgram);
 
+  // The function returns a vector of SYCL device images in required state,
+  // which are compatible with at least one of the device from Devs.
+  std::vector<device_image_plain>
+  getSYCLDeviceImages(const context &Ctx, const std::vector<device> &Devs,
+                      bundle_state State);
+
+  std::vector<device_image_plain>
+  getDeviceImages(const context &Ctx, const std::vector<device> &Devs,
+                  const std::vector<kernel_id> &KernelIDs, bundle_state State);
+
+  std::vector<device_image_plain>
+  getDeviceImages(const context &Ctx, const std::vector<device> &Devs,
+                  const DevImgSelectorImpl &Selector, bundle_state State);
```

Comment:
It seems that these functions are not needed since the kernel id and selector based filtering is handled in `kernel_bundle_impl` constructors.